src/NuGet.Core/NuGet.LibraryModel/LibraryTypeFlag.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.LibraryModel
+{
+    // Values are from LibraryTypes
+    // LibraryTypeConstraintFlag describes the types allowed to fill a dependency.
+    [Flags]
+    public enum LibraryTypeFlag : ushort
```

Comment:
`LibraryDependencyTarget` isn't a bad name, and it refers directly to the property name